external/vcm/vcm/calc/calc.py

```diff
@@ -43,7 +52,11 @@ def weighted_average(
 
 
 def vertical_tapering_scale_factors(n_levels: int, cutoff: int, rate: float):
-    z_arr = np.arange(n_levels)
-    scaled = np.exp((z_arr[slice(None, cutoff)] - cutoff) / rate)
-    unscaled = np.ones(n_levels - cutoff)
-    return np.hstack([scaled, unscaled])
+    if CUPY_AVAILABLE and cp.cuda.is_available():
```

Comment:
Since we don't need to check an input array, I would remove the `CUPY_AVAILABLE` and just do the check of `cuda.is_available()` on import and set `xp` to `np` or `cp` depending on that.